lib/routes/initial_walkthrough/initial_walkthrough.dart

```diff
@@ -162,7 +162,10 @@ class InitialWalkthroughPageState extends State<InitialWalkthroughPage>
 
     final themeProvider = ThemeProvider.controllerOf(context);
     try {
-      await accountBloc.connect(mnemonic: mnemonic ?? bip39.generateMnemonic(strength: 128));
+      await accountBloc.connect(
+        mnemonic: mnemonic ?? bip39.generateMnemonic(strength: 128),
```

Comment:
How about pass null in the mnemonics in `restore` case and then we don't need the `restored` flag?

Reply:
Account bloc uses the null case already: https://github.com/breez/c-breez/pull/664/files#diff-5dbe6ee35eb95a61a99689c83031840c5e21fc8f6a0f24b8b42ecbc2a87ab16bR54 when it is null, it doesn't change the current state of the validation status.
So, in order to use null instead of the boolean flag, I need to change how the bloc is started, from `if (!state.initial) connect();` to something like `if (!state.initial) init();` and adding this init method.
Let me know if you prefer this way Roei.